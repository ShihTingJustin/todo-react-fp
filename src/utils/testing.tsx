import React, { FC, ReactElement } from 'react';
import { Provider } from 'react-redux';

// import {  LocationProvider } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';

import { store } from '@Redux/store';

// const history = createHistory(window);
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render, AllTheProviders };
