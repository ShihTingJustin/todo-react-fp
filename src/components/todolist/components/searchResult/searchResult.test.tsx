import renderer from 'react-test-renderer';
import { AllTheProviders, render, screen } from '@Utils/testing';
import SearchResult from './searchResult';

describe('Test Button component', () => {
  const props = {
    keyword: 'qwe',
  };
  const component = AllTheProviders({ children: <SearchResult {...props} /> });

  it('[Render] match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('[Render] toBeInTheDocument', () => {
    render(component);
    const element = screen.getByTestId('searchResult');
    expect(element).toBeInTheDocument();
  });
});
