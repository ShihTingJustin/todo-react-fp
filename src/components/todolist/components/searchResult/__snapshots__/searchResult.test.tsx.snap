// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component [Render] match snapshot 1`] = `
<div
  data-testid="searchResult"
/>
`;
